bundle/edu.gemini.dataman.app/src/main/scala/edu/gemini/dataman/gsa/query/GsaQaUpdateQuery.scala

```diff
@@ -0,0 +1,37 @@
+package edu.gemini.dataman.gsa.query
+
+import edu.gemini.spModel.core.Site
+import edu.gemini.spModel.dataset.{DatasetQaState, DatasetLabel}
+
+import argonaut._
+import Argonaut._
+
+import java.net.URL
+
+/** A synchronous GSA query that sets the dataset QA state for one or more
+  * datasets. */
+sealed trait GsaQaUpdateQuery {
+  def setQaStates(updates: List[GsaQaUpdate.Request]): GsaResponse[List[GsaQaUpdate.Response]]
+
+  def setQaState(label: DatasetLabel, qa: DatasetQaState): GsaResponse[List[GsaQaUpdate.Response]] =
+    setQaStates(List(GsaQaUpdate.Request(label, qa)))
+}
+
+object GsaQaUpdateQuery {
+  import GsaQaUpdate._
+
+  /** Constructs a QA update query for the given host and site.  The `auth`
+    * parameter is how the GSA service provides some minimal protection against
+    * inappropriate setting of QA states.  The value is sent in a cookie that
```

Comment:
This is what I'm referring to by "moral dilemma" in the PR description.  It's just "known".  Specifically we'll keep it in svn in `OcsCredentials` along with the other sensitive information.  I'm sure we're doing worse things in our own codebase and this doesn't concern me too much, but I thought I should highlight it for discussion just in case.